client/src/App.js

```diff
@@ -1,11 +1,175 @@
+import Form from "react-bootstrap/Form";
+import InputGroup from "react-bootstrap/InputGroup";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Video from "./Video";
 
 function App() {
+  const [title, setTitle] = useState("");
+  const [url, setUrl] = useState("");
+  const [items, setItems] = useState(null);
+  const [search, setSearch] = useState("");
+  const [isOpen, setIsOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isAscending, setIsAscending] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await axios.get(
+          `https://full-stack-assessment.onrender.com/videos?ordering=${
+            isAscending ? "asc" : "desc"
+          }`
+        );
+        setIsLoading(true);
+        setItems(response.data);
+      } catch (e) {
+        console.log(e);
+      }
+    };
+
+    fetchData();
+  }, [isAscending]);
+
+  const orderHandler = (e) => {
+    setIsAscending(!isAscending);
+  };
+
+  const searchHandler = async () => {
+    const res = await axios.get(
+      `https://full-stack-assessment.onrender.com/search/?title=${search}`
+    );
+    setItems(res.data);
+  };
+
+  const openHandler = () => {
+    setIsOpen(true);
+  };
+
+  const cancelHandler = () => {
+    setIsOpen(false);
+  };
+
+  const urlHandler = (event) => {
+    setUrl(event.target.value);
+  };
+
+  const titleHandler = (event) => {
+    setTitle(event.target.value);
+  };
+
+  const validateUrl = (value) => {
+    // Regular expression for URL validation
+    const urlRegex = /^(ftp|http|https):\/\/[^ "]+$/;
+    if (!value || !urlRegex.test(value)) {
+      alert("Please enter a valid URL");
+    }
+    return null; // Return null for no validation errors
+  };
+
+  const addClickHandler = async () => {
+    if (!title) {
+      alert("Please enter a title");
+      return;
+    }
+    validateUrl(url);
+    try {
+      const response = await axios.post(
+        "https://full-stack-assessment.onrender.com/video",
+        {
+          title: title.toLowerCase(),
+          url: url.split("watch?v=")[1],
+        }
+      );
+      setTitle("");
+      setUrl("");
+      alert("Video has been sent");
+      window.location.reload();
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
   return (
     <div className="App">
-      <header className="App-header">
-        <h1>Video Recommendation</h1>
+      <header className="App-header bg-primary py-4 custom-header">
+        <h1 className="text-center mb-0">Video Recommendation</h1>
       </header>
+
+      <Container className="p-4">
+        <Row>
+          <Col md={4} className="mb-4">
+            <Button variant="link" onClick={openHandler}>
+              Add Video
+            </Button>
+            {isOpen && (
+              <>
+                <InputGroup className="mb-3">
```

Comment:
It's good for accessibility to wrap these inputs in a `form` element and connect its `onSubmit` event to `addClickHandler` instead of the button `onClick`. Then users can submit the form by pressing enter while in an input; also screen reader users will get more useful information.

Reply:
I will